src/components/UI/alert.tsx

```diff
@@ -0,0 +1,173 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import {
+  AnnotationIcon,
+  CheckIcon,
+  ExclamationIcon,
+  InformationCircleIcon,
+  ShieldExclamationIcon,
+  XIcon
+} from '@heroicons/react/outline'
+
+import { AlertContentProps, AlertVariant } from '@/context/AlertContext'
+
+const iconColor = '#000000'
+const titleColor = '#000000'
+const textColor = '#000000'
+
+const shadow =
+  '0 2px 4px rgba(0,0,0,0.1),\
+                0 4px 8px rgba(0,0,0,0.1),\
+                0 8px 16px rgba(0,0,0,0.1),\
+                0 16px 32px rgba(0,0,0,0.1),\
+                0 32px 64px rgba(0,0,0,0.1)'
+
+const buttonClasses =
+  'mx-auto mt-2 w-fit rounded-lg bg-primary py-1 px-4 text-lg text-white shadow-md focus:outline-primary active:bg-dark-red'
+const getContainerClasses = (vis: boolean, pos: string) => {
+  return (
+    `flex items-center fixed right-0 left-0 z-999 pointer-events-auto bg-white text-black border-x-1 border-[#dddddd] transition-transform duration-700` +
+    `${
+      pos === 'top'
+        ? ' rounded-b-md border-b-1 -top-[250px] pl-5 pr-4 py-3 mx-[1rem]' +
+          (vis ? ' translate-y-[250px]' : ' translate-y-[0px]')
+        : pos === 'bottom' &&
+          ' rounded-t-md border-t-1 bottom-[-300px] p-5 mx-[3rem] flex-wrap justify-center' +
+            (vis ? ' translate-y-[-300px]' : ' translate-y-[0px]')
+    }`
+  )
+}
+
+type AlertProps = {
+  visible: boolean
+  setVisible: (visible: boolean) => void
+  content: AlertContentProps
+}
+
+const Alert: React.FC<AlertProps> = ({ visible, setVisible, content }) => {
+  const [contentCache, setContentCache] = useState<AlertContentProps>({
```

Comment:
Is a content cache really needed? A bit confused why the component can't directly use the `content` prop. If `content` changes then the whole component re-renders so `contentCache` would get recreated anyway (from my understanding).

Reply:
See above :)) appreciate the review.

Reply:
resolved prematurely, the reason this is used is in case an alert is going to be called twice. If the content is the same no update is called so we need to clear the content. This means a cache is required so while the out animation plays it will still display the previous alert. Another solution I could implement would be a timer for the animation duration clearing the content after the timer ends. If that is better let me know.

Reply:
Right that makes sense, I think you're idea is fine as is then